packages/libs/react-ui/src/components/NavHeader/NavHeaderSelect.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  chevronIconClass,
+  selectClass,
+  selectContainerClass,
+  selectContainerClassDisabled,
+  selectIconClass,
+} from './NavHeader.css';
+
+import { SystemIcon } from '@components/Icon';
+import classNames from 'classnames';
+import type { FC } from 'react';
+import React, { forwardRef } from 'react';
+
+export interface INavHeaderSelectProps
+  extends Omit<
+    React.HTMLAttributes<HTMLSelectElement>,
+    'aria-label' | 'as' | 'className' | 'children' | 'id'
+  > {
+  ariaLabel: string;
+  children: React.ReactNode;
+  disabled?: boolean;
+  icon?: keyof typeof SystemIcon;
+  ref?: React.ForwardedRef<HTMLSelectElement>;
+  onChange?: React.ChangeEventHandler<HTMLSelectElement>;
+  id: string;
+  value?: string;
+}
+
+export const NavHeaderSelect: FC<INavHeaderSelectProps> = forwardRef<
+  HTMLSelectElement,
+  INavHeaderSelectProps
+>(function Select(
+  { ariaLabel, children, disabled = false, onChange, icon, ...rest },
+  ref,
+) {
+  const Icon = icon && SystemIcon[icon];
+  const ChevronDown = SystemIcon.ChevronDown;
+
+  return (
+    <div
+      className={classNames(selectContainerClass, {
+        [selectContainerClassDisabled]: disabled,
+      })}
+    >
+      {Icon && (
+        <span className={selectIconClass}>
+          <Icon size="md" />
+        </span>
+      )}
+      <select
+        aria-label={ariaLabel}
+        className={selectClass}
+        disabled={Boolean(disabled)}
+        ref={ref}
+        {...rest}
```

Comment:
I don't think we should spread the rest here. I see that it's the same with the original Select component we have and think we should remove this line from both.

Also, if we do spread this in, check your console and you should see a warning related to setting the value without an onChange handler.

Reply:
mmm I do think we would want to pass the native select props to the select components. Since this is a thin style wrapper around a native select element, I don't think it should take away any native functionality. The warning should be resolved though, I think maybe we can remove the onChange prop in the types or something since it will be included with the native select element.

Reply:
I will remove the onChange prop in the type and use it from the native select